pkg/nodescheduler/resourcemanager_test.go

```diff
@@ -0,0 +1,38 @@
+package nodescheduler
+
+import (
+	"reflect"
+	"regexp"
+	"testing"
+)
+
+func TestEnvFromSecret(t *testing.T) {
+	pluginEnvFromSecretRegexFormat = regexp.MustCompile(`^{secret\.([a-z0-9-]+).([a-zA-Z0-9]+)}$`)
```

Comment:
It seems like this test is actually doing a reimplementation of the logic above and testing that rather than testing the main implementation. For example, if we were to change the way this worked internally, this test would still pass as far as I can tell.

It'd be nice if we instead have a  unit test which just verifies the Kubernetes YAML for a submission that contains a secrets field.

Reply:
Right. I totally agree with it. In fact, it served me as a function for me to quickly prototype how to implement it, rather than as a real unit test for the function, which even didn't exist. Since we have some time before we ask people to try this. I will see if I can make the function better to be unit tested. Thanks for the comment.

Reply:
This is totally ok, too. We can treat this as in "internal" or "scaffolding" unit test to get the details right. I think the most important thing is to get feedback and see if the interface needs to change at all.

The simplest thing to do in there short term is extract that logic into a function which the implementation uses too. Longer term, we can drop it in place of checking the actual YAML output.

Reply:
Alright. I improved the unit test by employing the Kubernetes fake Clientset object that allows switching between real and fake Kubernetes backends. With it, we can use the ResourceManager object on both backends without modifying or adding any in the class.